source/Assets/Plugins/Didomi/IOS/Didomi.mm

```diff
@@ -691,7 +768,7 @@ void onReady(callback_function pFunc)
 
 
 
-void addEventListener( void (*event_listener_handler) (DDMEventType, NSString * _Nullable ))
+void addEventListener( void (*event_listener_handler) (int, NSString * _Nullable ))
```

Comment:
why is this type changing from DDMEventType to int?

Reply:
in the onError event there is no EventType defined like in the other events as you can see in the code below.

```swift
@objc public var onReady: (EventType) -> Void = { event in }
    @objc public var onError: (DidomiErrorEvent) -> Void = { error in }
    @objc public var onShowNotice: (EventType) -> Void = { event in }
    @objc public var onNoticeClickAgree: (EventType) -> Void = { event in }
```
Also there is no enum value that can be used for onError event. Because EventType is not used for onError. You can see the enum values below.

typedef SWIFT_ENUM_NAMED(NSInteger, DDMEventType, "EventType", open) {
  DDMEventTypeConsentChanged = 0,
  DDMEventTypeHideNotice = 1,
  DDMEventTypeReady = 2,
  DDMEventTypeShowNotice = 3,
  DDMEventTypeNoticeClickAgree = 4,
  DDMEventTypeNoticeClickDisagree = 5,
  DDMEventTypeNoticeClickMoreInfo = 6,
  DDMEventTypeNoticeClickViewVendors = 7,
  DDMEventTypeNoticeClickPrivacyPolicy = 8,
  DDMEventTypePreferencesClickAgreeToAll = 9,
  DDMEventTypePreferencesClickDisagreeToAll = 10,
  DDMEventTypePreferencesClickAgreeToAllPurposes = 11,
  DDMEventTypePreferencesClickDisagreeToAllPurposes = 12,
  DDMEventTypePreferencesClickResetAllPurposes = 13,
  DDMEventTypePreferencesClickAgreeToAllVendors = 14,
  DDMEventTypePreferencesClickDisagreeToAllVendors = 15,
  DDMEventTypePreferencesClickPurposeAgree = 16,
  DDMEventTypePreferencesClickPurposeDisagree = 17,
  DDMEventTypePreferencesClickCategoryAgree = 18,
  DDMEventTypePreferencesClickCategoryDisagree = 19,
  DDMEventTypePreferencesClickViewVendors = 20,
  DDMEventTypePreferencesClickViewPurposes = 21,
  DDMEventTypePreferencesClickSaveChoices = 22,
  DDMEventTypePreferencesClickVendorAgree = 23,
  DDMEventTypePreferencesClickVendorDisagree = 24,
  DDMEventTypePreferencesClickVendorSaveChoices = 25,
  DDMEventTypeSyncDone = 26,
};


To bridge it between the IOS SDK and Unity SDK, type of the input changed from DDMEventType to the int. And the value 1000 is used for onError event type value.